mp/src/game/server/EnvFade.cpp

```diff
@@ -108,6 +113,47 @@ void CEnvFade::InputFade( inputdata_t &inputdata )
 	m_OnBeginFade.FireOutput( inputdata.pActivator, this );
 }
 
+//-----------------------------------------------------------------------------
+// Purpose: Input handler that does the reverse screen fade.
+//-----------------------------------------------------------------------------
+void CEnvFade::InputFadeReverse(inputdata_t &inputdata)
+{
+    int fadeFlags = 0;
+
+    if (m_spawnflags & SF_FADE_IN)
+    {
+        fadeFlags |= FFADE_OUT;
+    }
+    else
+    {
+        fadeFlags |= FFADE_IN;
+    }
+
+    if (m_spawnflags & SF_FADE_MODULATE)
+    {
+        fadeFlags |= FFADE_MODULATE;
+    }
+
+    if (m_spawnflags & SF_FADE_STAYOUT)
+    {
+        fadeFlags |= FFADE_STAYOUT;
+    }
+
+    if (m_spawnflags & SF_FADE_ONLYONE)
+    {
+        if (inputdata.pActivator && inputdata.pActivator->IsNetClient())
+        {
+            UTIL_ScreenFade(inputdata.pActivator, m_clrRender, ReverseDuration(), HoldTime(), fadeFlags);
+        }
+    }
+    else
+    {
+        UTIL_ScreenFadeAll(m_clrRender, ReverseDuration(), HoldTime(), fadeFlags | FFADE_PURGE);
+    }
+
+    m_OnBeginFade.FireOutput(inputdata.pActivator, this);
+}
+
```

Comment:
This code also looks very similar to the other InputFade, you should pull it out into a shared private method that they both call (use a bool to determine if reverse)